app/assets/javascripts/discourse/app/mixins/card-contents-base.js

```diff
@@ -86,7 +86,9 @@ export default Mixin.create({
     });
 
     this.appEvents.trigger("user-card:show", { username });
-    this._showCallback(username, $(target));
+    this._showCallback(username, $(target)).then(() => {
+      this.appEvents.trigger("user-card:after-show", { username });
```

Comment:
I'm wondering if it is better if we just return the user object instead as we'll get to support more use cases.

Reply:
good call, I'll update this tomorrow :+1: